vumi/transports/smpp/smpp_transport.py

```diff
@@ -298,6 +298,15 @@ def _reject_for_invalid_address(self, message, field):
         return self.publish_nack(
             message['message_id'], u'Invalid %s: %s' % (field, message[field]))
 
+    def on_connection(self):
+        return self.publish_status_unbound()
+
+    def publish_status_unbound(self):
+        return self.publish_status('smpp', 'major', reasons=[{
+            'type': 'unbound',
+            'message': 'Connected but not bound',
+        }])
```

Comment:
Wondering if there are any `details` we should add. Thought about adding the host and port details, but that is a thing that should already be known by the thing that created the transport. In Junebug's case, the client using the Junebug API should know host and port details since it is the thing that configured the transport to use those details.